components/org.wso2.micro.integrator.initializer/src/main/java/org/wso2/micro/integrator/initializer/utils/ServiceCatalogUtils.java

```diff
@@ -372,6 +388,11 @@ private static boolean processMetadata(File tempDir, File metadataFolder, File m
                                            Map<String, String> md5MapOfAllService) throws IOException
             , ResolverException, NoSuchAlgorithmException {
         String metaFileName = metadataYamlFolder.getName();
+        // check metadata is relevant to proxy-service
+        if (metaFileName.contains(PROXY_SERVICE_SUFFIX) && !(new File(metadataFolder,
+                metaFileName.replaceAll(METADATA_FOLDER_STRING, SWAGGER_FOLDER_STRING))).exists()) {
+            return processProxyServiceMetadata(tempDir, metadataYamlFolder, md5MapOfAllService);
+        }
         String APIName = metaFileName.substring(0, metaFileName.indexOf(METADATA_FOLDER_STRING));
```

Comment:
What if the metadataFileName contains the PROXY_SERVICE_SUFFIX and the swagger folder exists? Are we going to treat it as an API? IMO, this is wrong, we should not allow creation of such a proxy in the integration studio

Reply:
Here, it checks the swagger **file** exists inside the metadata folder. Folder structures for the APIs and Proxy services will be as follows.

**If the folder is for API metadata:**
- APIName_metadata_1.0.0
      -  APIName_metadata-1.0.0.yaml
      -  APIName_swagger-1.0.0.yaml

**If the folder is for Proxy metadata:**
- ProxyName_proxy_metadata_1.0.0
      -  ProxyName_proxy_metadata-1.0.0.yaml